Sources/MessagingInApp/Gist/Managers/MessageManager.swift

```diff
@@ -60,6 +68,16 @@ class MessageManager: EngineWebDelegate {
         }
     }
 
+    func cancelShowingMessage() {
```

Comment:
I do agree, better threading support in the in-app SDK is needed. I also agree this is out-of-scope because I don't think there is a quick win for MessageManager, for example, synchronized when the rest of the module could also be improved. 

I think bigger module-wide refactors could be a more productive approach to this problem. I also think that swift concurrency could solve this problem really well. 

Do you disagree and think there is a quick win here? Is it a blocker?